src/aker_metrics.c

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright 2021 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdbool.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "aker_metrics.h"
+#include "aker_log.h"
+
+static aker_metrics_t *g_metrics = NULL;
+
+aker_metrics_t* get_global_metrics(void)
+{
+	aker_metrics_t* tmp = NULL;
+	tmp = g_metrics;
+	return tmp;
+}
+
+int init_global_metrics()
+{
+	aker_metrics_t* metrics;
+	metrics = (aker_metrics_t*)malloc(sizeof(aker_metrics_t));
+	
+	if(metrics)
+	{
+		memset(metrics, 0, sizeof(aker_metrics_t));
+
+		metrics->device_block_count = 0;
+		metrics->windows_transistion_count = 0;
+		metrics->schedule_set_count = 0;
+		metrics->md5_error_count = 0;
+		metrics->process_start_time = 0;
+		metrics->schedule_enabled = 0;
+		metrics->timezone = strndup("NULL", strlen("NULL"));
+
+	}
+	else
+	{
+		debug_error("malloc failed in metrics\n");
+		return 0;
+	}
+
+	if(g_metrics == NULL)
+	{
+		g_metrics = metrics;
+	}
+
+	return 1;
+}
+int set_aker_metrics(int metrics,int num, ... )
```

Comment:
Make this into 7 different functions.  While this will work, it is error prone.

```c
void aker_metric_inc_device_block_count( uint32_t val );
void aker_metric_inc_window_trans_count( uint32_t val );
void aker_metric_inc_schedule_set_count( uint32_t val );
void aker_metric_inc_md5_err_count( uint32_t val );
void aker_metric_set_process_start_time( time_t val );
void aker_metric_set_tz( const char *val );
```

Do add a mutex that is grabbed & released inside each call.  Otherwise you have race conditions.